site/walkthrough.md

```diff
@@ -0,0 +1,109 @@
+---
+title: weave Walkthrough
+---
+
+## launch DNS by default, IP allocation and naming, both with sensible defaults, rendezvous peering, auto-attach
+
+````bash
+root@a:~# weave launch
+root@a:~# weave run --name=pong -d -ti ubuntu
+````
+
+Weave automatically picked an IP range that isn't used locally (this
+range can also be specified explicitly, as an argument to `weave
+launch`), and allocated the container an IP in that range. The
+container is registered in weaveDNS as `pong.weave.local`. weaveDNS
+was launched automatically since no `--no-dns` option was given on
+`weave launch`.
+
+````bash
+root@b:~# weave launch
+root@b:~# weave run -ti ubuntu
+root@c0c8b4a7d709:/# ping pong
+ping pong.weave.local (10.0.1.1) 56(84) bytes of data.
+64 bytes from pong.weave.local (10.0.1.1): icmp_seq=1 ttl=64 time=0.036 ms
+64 bytes from pong.weave.local (10.0.1.1): icmp_seq=2 ttl=64 time=0.031 ms
+^C
+--- pong.weave.local ping statistics ---
+2 packets transmitted, 2 received, 0% packet loss, time 999ms
+rtt min/avg/max/mdev = 0.031/0.033/0.036/0.006 ms
+````
+
+Weave peered with A through rendezvous (since both peers happened to
+be running in some local network), learned from A what the IP range
+is, checked that it is unused locally (it would complain if it was),
+allocated an IP from that range for the container, configured name
+resolution to go to weaveDNS and search in the weave.local
+domain. Note that the container is not registered in DNS since it
+wasn't explicitly named.
+
+We didn't specify `-d` in `weave run` and therefore, just like in
+`docker run`, automatically attached to the container.
+
+## load balancing
+
+````bash
+root@c:~# weave launch
+root@c:~# weave run --name=pong -d -ti ubuntu
+````
+
+````bash
+root@c0c8b4a7d709:/# ping pong
+ping pong.weave.local (10.0.1.1) 56(84) bytes of data.
+64 bytes from pong.weave.local (10.0.1.1): icmp_seq=1 ttl=64 time=0.036 ms
+64 bytes from pong.weave.local (10.0.1.1): icmp_seq=2 ttl=64 time=0.031 ms
+^C
+--- pong.weave.local ping statistics ---
+2 packets transmitted, 2 received, 0% packet loss, time 999ms
+rtt min/avg/max/mdev = 0.031/0.033/0.036/0.006 ms
+root@c0c8b4a7d709:/# ping pong
+ping pong.weave.local (10.0.1.3) 56(84) bytes of data.
+64 bytes from pong.weave.local (10.0.1.3): icmp_seq=1 ttl=64 time=0.036 ms
+64 bytes from pong.weave.local (10.0.1.3): icmp_seq=2 ttl=64 time=0.031 ms
+^C
+--- pong.weave.local ping statistics ---
+2 packets transmitted, 2 received, 0% packet loss, time 999ms
+rtt min/avg/max/mdev = 0.031/0.033/0.036/0.006 ms
+````
+
+The container we started on `c` has the same name as the container we
+started on `a`. As a result, name resolution will return the addresses
+of both containers in random order.
+
```

Comment:
Just to note that, while starting two containers called 'pong' on different machines is allowed, Docker won't let you run two containers on the same machine with the same name, so you can't load-balance in this way across multiple processes without caring where they run.

Maybe in the context of swarm, Docker would even disallow the multi-machine case.

So, it may be sub-optimal to recommend this as a primary mechanism.